README.md

```diff
@@ -2,18 +2,195 @@
 
-To get started with Layer0, see [docs.xfra.ims.io](https://quintilesims.github.io/layer0/).
+To get started with Layer0, see [quintilesims.github.io/layer0](https://quintilesims.github.io/layer0/).
 
-## Layer0 Developers
+# Components
+The following components comprise the Layer0 Application. 
 
-### Tools in this Repo
+## Layer0 Setup
+The Layer0 Setup tool (commonly called `l0-setup`) is used to provision and manage resources in AWS that a Layer0 instance needs to run correctly. 
 
-* Layer0 API source is located in `/api`. For developer notes, see [/api/README.md](/api/README.md).
-* Layer0 Setup source (`l0-setup`) is located in `/setup`. For developer notes, see [/setup/README.md](/setup/README.md).
-* Layer0 CLI source (`l0`) is located in `/cli`. For developer notes, see [/cli/README.md](/cli/README.md).
-* Layer0 Runner source is located in `/runner`. For developer notes, see [/runner/README.md](/runner/README.md).
-* Layer0 Documentation is located in `/docs`. For developer notes, see [/docs/README.md](/docs/README.md).
+#### Common Functions
+*The following commands should be ran from the `layer0/setup` directory*
 
-### Releasing a New Version
+| Function      | Command       |
+| ------------- |:--------------|
+| Run program   | `go run main.go` |
+| Build binary  | `go build -o l0-setup` |
+| Cross-Compile | `make build` |
+| Release       | `make release` |
 
+## Layer0 CLI
+The Layer0 CLI (commonly called `l0`) is used to manage Layer0 resources from the shell.
+
+#### Common Functions
+*The following commands should be ran from the `layer0/cli` directory*
+
+| Function      | Command       |
+| ------------- |:--------------|
+| Run program   | `go run main.go` |
+| Build binary  | `go build -o l0` |
+| Cross-Compile | `make build` |
+| Release       | `make release` |
+
+## Layer0 API
+The Layer0 API is a web service that provisions and manages Layer0 resources in AWS.
+
+#### Common Functions
+*The following commands should be ran from the `layer0/api` directory*
+
+| Function      | Command       |
+| ------------- |:--------------|
+| Run program        | `go run main.go` |
+| Build binary       | `go build -o l0-api` |
+| Build Docker Image | `make build` |
+| Release            | `make release` |
+
+## Layer0 Runner
+The Layer0 Runner is a service that runs jobs created by the Layer0 API.
+
+#### Common Functions
+*The following commands should be ran from the `layer0/runner` directory*
+
+| Function      | Command       |
+| ------------- |:--------------|
+| Run program        | `go run main.go` |
+| Build binary       | `go build -o l0-runner` |
+| Build Docker Image | `make build` |
+| Release            | `make release` |
+
+## Docs
+The documentation for Layer0 exists in `layer0/docs-src`. 
+We use [mkdocs](http://www.mkdocs.org/) to compile our markdown docs into html and css.
```

Comment:
We currently require `mkdocs` `>v0.16.0`. Do we need to make that explicit here? It shouldn't be a problem if a new dev uses `/docs-src/make deps` (installing latest), but a system with an older pre-installed version of `mkdocs` could run into a problem.